library/lua/gui/widgets.lua

```diff
@@ -433,6 +438,47 @@ function Label:setText(text)
         self.frame = self.frame or {}
         self.frame.h = self:getTextHeight()
     end
+
+    self._text_dirty = true
+end
+
+function Label:update_scroll_inset()
+    if self.show_scroll_icons == nil then
+        self._show_scroll_icons = self:getTextHeight() > self.frame_body.height and 'right' or false
+    else
+        self._show_scroll_icons = self.show_scroll_icons
+    end    
+    if self._show_scroll_icons then
+        -- here self._show_scroll_icons can only be either
+        -- 'left' or any true value which we interpret as right
+        local l,t,r,b = gui.parse_inset(self.frame_inset)
+        if self._show_scroll_icons == 'left' and l <= 0 then
+            l = 1
+        elseif r <= 0 then
+            r = 1
+        end
+        self.frame_inset = {l=l,t=t,r=r,b=b}
+    end
+end
+
+function Label:render_scroll_icons(dc,rect)
+    if self:getTextHeight() > self.frame_body.height then
+        local x = self._show_scroll_icons == 'left' and 0 or dc.width-1
+        if self.start_line_num ~= 1 then
+            dc:seek(x, 0):char(self.up_arrow_icon, self.scroll_icon_pen)
+        end
+        local last_visible_line = self.start_line_num + self.frame_body.height - 1
+        if last_visible_line < self:getTextHeight() then
+            dc:seek(x, dc.height-1):char(self.down_arrow_icon, self.scroll_icon_pen)
+        end
+    end
+end
+
+function Label:postComputeFrame()
+    if self._text_dirty then
```

Comment:
did that too. I forgot the first rule of optimization: don't optimize.

Reply:
Also, the scroll doesn't work with MenuOverlay (at least the way how I tested it), because arrow keys move the cursor on the map... But that is out of scope of this pull request.